### DISCUSSION TITLE
Poi(Point Of Interest) display error using tfs (osgText)

### DISCUSSION THREAD

**Discussion by tjdtnfksp** [pelicanmapping/osgearth#2013] [Support Q&A]
Can I ask you a question about the last version of osgearth?

Use OpenSceneGraph-3.6.4 + osgearth-2.10.2.

Use TFS to develop POI (Point Of Interest) functions.

I did the tiling, too.

And using minRange and MaxRange, it appears on the map by altitude, is erased (Movements zoom in and zoom out), and when repeated, the program stops.

I can't modify the source using dll.

Did I build something wrong with osgText?

Is it because there are a lot of poi displayed at once?

Please help me.

///my code

	osgEarth::Drivers::TFSFeatureOptions featureOptions;
	featureOptions.url() = url;
	featureOptions.format() = "json";

	Style poiStyle;

	TextSymbol* text = poiStyle.getOrCreateSymbol<TextSymbol>();
	text->content() = StringExpression("[eng_nm]");
	//text->priority() = NumericExpression("[ID]");
	text->size() = 16.0f;
	text->alignment() = TextSymbol::ALIGN_CENTER_CENTER;
	text->fill()->color() = Color::White;
	text->halo()->color() = Color::DarkGray;
	text->encoding() = osgEarth::Symbology::TextSymbol::ENCODING_UTF8;
	text->declutter() = false;

	StyleSheet* styleSheet = new StyleSheet();
	styleSheet->addStyle(poiStyle);
	
	osgEarth::Drivers::FeatureGeomModelOptions geomOptions;
	geomOptions.featureOptions() = featureOptions;
	geomOptions.styles() = styleSheet;
	geomOptions.enableLighting() = false;	
	
	geomOptions.maxRange() = max;
	geomOptions.minRange() = min;

	FeatureModelLayer* layer = new FeatureModelLayer(geomOptions);	
	mMap->addLayer(layer);

![1](https://user-images.githubusercontent.com/50495880/157806366-23533910-4336-4a8d-ba7d-d86ddca09797.png)
![2](https://user-images.githubusercontent.com/50495880/157806373-b8ca6e6f-007c-4d27-8489-c08f96cbc73d.png)
![3](https://user-images.githubusercontent.com/50495880/157806387-740ccc5c-af5e-4fa2-bd56-3c2b4fd4a99c.png)

**Comment by gwaldron** [comment]
If I had to guess, I would say you encountered a thread-safety issue in the osgText library. Unfortunately I don't have a good workaround to suggest. I once thought about pre-loading all the text glyphs so they don't get created on the fly; that would probably solve the problem, but we never looked into it further.
